tests/cxx/pragma_associated.cc

```diff
@@ -0,0 +1,25 @@
+//===--- pragma_associated.cc - test input file for iwyu ------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "tests/cxx/pragma_associated-d1.h"  // IWYU pragma: associated
```

Comment:
My motivation for testing multiple files is that we change `associated_pragma_location_` state (valid/invalid) in multiple places. And that is somewhat error-prone.